bundle/edu.gemini.qpt.client/src/main/java/edu/gemini/qpt/core/listeners/LimitsListener.java

```diff
@@ -153,20 +205,25 @@ public void propertyChange(PropertyChangeEvent evt) {
 				double minDelta = Math.abs(haLimitMin - minHA);
 				double maxDelta = Math.abs(haLimitMax - maxHA);
 
+				Option<Marker> haMarker = None.instance();
 				if (minHA < haLimitMin) {
-					mm.addMarker(false, this, Severity.Error, String.format("Hour angle constraint violated (%s < %s).", TimeUtils.hoursToHHMMSS(minHA), TimeUtils.hoursToHHMMSS(haLimitMin)), v, a);
+					haMarker = new Some<>(new Marker(false, this, Severity.Error, String.format("Hour angle constraint violated (%s < %s).", TimeUtils.hoursToHHMMSS(minHA), TimeUtils.hoursToHHMMSS(haLimitMin)), v, a));
 				} else if (minDelta < 1. / 15.) {
-					mm.addMarker(false, this, Severity.Warning, "Target comes within 1\u00B0 of lower HA constraint.", v, a);
+					haMarker = new Some<>(new Marker(false, this, Severity.Warning, "Target comes within 1\u00B0 of lower HA constraint.", v, a));
 				}
 
 				if (maxHA > haLimitMax) {
-					mm.addMarker(false, this, Severity.Error, String.format("Hour angle constraint violated (%s > %s).", TimeUtils.hoursToHHMMSS(maxHA), TimeUtils.hoursToHHMMSS(haLimitMax)), v, a);
+					haMarker = new Some<>(new Marker(false, this, Severity.Error, String.format("Hour angle constraint violated (%s > %s).", TimeUtils.hoursToHHMMSS(maxHA), TimeUtils.hoursToHHMMSS(haLimitMax)), v, a));
 				} else if (maxDelta < 1. / 15.) {
-					mm.addMarker(false, this, Severity.Warning, "Target comes within 1\u00B0 of upper HA constraint.", v, a);
+					haMarker = new Some<>(new Marker(false, this, Severity.Warning, "Target comes within 1\u00B0 of upper HA constraint.", v, a));
 				}
 
-				break;
+				haMarker.orElse(() -> {
+					final String m = String.format("Must be observed in the hour angle range %1.2f - %1.2f", haLimitMin, haLimitMax);
+					return createSolverMarker.apply("VisibleUnionCache", m);
```

Comment:
Here we have a capital `V` in `VisibleUnionCache` which underscores the need to put these names in constants.